ScoreSaber/UI/Leaderboard/ScoreSaberLeaderboardViewController.cs

```diff
@@ -1,77 +1,64 @@
 using ScoreSaber.Core.Data.Models;
 using ScoreSaber.Core.Data.Wrappers;
 using ScoreSaber.Core.ReplaySystem;
+using ScoreSaber.Core.ReplaySystem.Data;
 using ScoreSaber.Core.Services;
+using ScoreSaber.Core.Utils;
 using ScoreSaber.Extensions;
 using ScoreSaber.UI.Elements.Leaderboard;
 using ScoreSaber.UI.Elements.Profile;
+using ScoreSaber.UI.Leaderboard;
+using ScoreSaber.UI.Main;
+using ScoreSaber.UI.PromoBanner;
+using SiraUtil.Affinity;
+using SiraUtil.Logging;
 using System;
+using System.Collections;
 using System.Collections.Generic;
-using System.Reflection;
+using System.Linq;
+using System.Net;
+using System.Threading;
 using System.Threading.Tasks;
+using TMPro;
 using UnityEngine;
+using UnityEngine.Diagnostics;
+using UnityEngine.EventSystems;
 using UnityEngine.UI;
 using Zenject;
-using ScoreSaber.Core.Utils;
-using ScoreSaber.Core.ReplaySystem.Data;
-using System.Threading;
+using static ScoreSaber.UI.Leaderboard.ScoreSaberLeaderboardViewController;
+using static System.Windows.Forms.VisualStyles.VisualStyleElement;
+using Button = UnityEngine.UI.Button;
 
 namespace ScoreSaber.UI.Leaderboard {
 
-    internal class ScoreSaberLeaderboardViewController : IInitializable, IDisposable {
-
-        #region BSML Components
-        [UIComponent("root")]
-        protected readonly RectTransform _root = null;
-        [UIParams]
-        private readonly BSMLParserParams _parserParams = null;
-        [UIComponent("up-button")]
-        protected readonly Button _upButton = null;
-        [UIComponent("down-button")]
-        protected readonly Button _downButton = null;
-
-        [UIValue("imageHolders")]
-        internal List<ProfilePictureView> _ImageHolders = null;
-        [UIValue("cellClickerHolders")]
-        internal List<CellClickingView> _cellClickingHolders = null;
-        [UIValue("entry-holder")]
-        internal readonly EntryHolder _infoButtons = null;
-        [UIValue("score-detail-view")]
-        protected readonly ScoreDetailView _scoreDetailView = null;
-        [UIComponent("profile-detail-view")]
-        protected readonly ProfileDetailView _profileDetailView = null;
-
-        [UIAction("up-button-click")] private void UpButtonClicked() => DirectionalButtonClicked(false);
-        [UIAction("down-button-click")] private void DownButtonClicked() => DirectionalButtonClicked(true);
-
-#if PPV3
-        [UIAction("PPv3-replay-click")] private void PPv3ReplayClick() => _ = PPv3ReplayClicked();
-#endif
-        #endregion
+    [HotReload(RelativePathToLayout = @"./ScoreSaberLeaderboardViewController.bsml")]
+    [ViewDefinition("ScoreSaber.UI.Leaderboard.ScoreSaberLeaderboardViewController.bsml")]
+    internal class ScoreSaberLeaderboardViewController : BSMLAutomaticViewController, INotifyLeaderboardSet, IInitializable {
 
-        public bool activated { get; private set; }
-        public int leaderboardPage { get; set; } = 1;
+        // TODO: Put both of these somewhere nicer?
```

Comment:
if really needed put these exactly around the corresponding elements with the corresponding pragrma warning restore afterwards.

Reply:
fixed